.github/workflows/wave.yml

```
@@ -0,0 +1,87 @@
+name: Generate conda-lock
+# When environment.yml is changed
+on:
+  pull_request:
+    paths:
+      - "**/environment.yml"
+
+jobs:
+  gen-matrix:
+    name: generate-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+
+      - name: Calculate file differences
+        id: diff
+        uses: tj-actions/changed-files@v44
+        with:
+          json: true
+          quotepath: false
+          files: |
+            modules/**/environment.yml
+      - name: Debug
+        run: echo ${{ steps.diff.outputs.all_changed_files }}
+      - id: set-matrix
+        run: echo "matrix={\"files\":${{ steps.diff.outputs.all_changed_files }} }" >> "$GITHUB_OUTPUT"
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  lock:
+    if: "${{ fromJson(needs.gen-matrix.outputs.matrix) }}"
+    needs: gen-matrix
+    name: Generate conda-lock file
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: "${{ fromJson(needs.gen-matrix.outputs.matrix) }}"
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-name: conda-lock-env
+          create-args: >-
+            python=3.11
+            conda-lock
+            mamba
+
+      - name: Get the lock file path
+        uses: actions/github-script@v7
+        id: lockfile-name
+        with:
+          result-encoding: string
+          script: |
+            return '${{ matrix.files }}'.replace('/environment.yml', '/conda-lock.yml');
+
+      - name: Run conda-lock
+        run: |
+          rm --force "${{steps.lockfile-name.outputs.result}}"
+          conda-lock lock $MAMBA --file "${{ matrix.files }}" --kind lock --platform linux-64 --lockfile "${{steps.lockfile-name.outputs.result}}"
```

```suggestion
          conda-lock lock $MAMBA --file "${{ matrix.files }}" --kind explicit --platform linux-64 --lockfile "${{steps.lockfile-name.outputs.result}}"
          mv  conda-linux-64.lock conda-lock-linux-64.yml
```

Nextflow needs it to be a .yml, but use of `--kind explicit` means we get .lock files, so we'll need to do some file renaming.

Platforms should probably be set in the environment.ymls actually.

---

Doesn't look like that works 😞

---

The `explicit` thing is important. It makes you a lock file for each platform which is consumable by `conda env create`, and therefore by Nextflow.

But by doing `explicit` you lose control over the naming of the multiple outputs, and Nextflow needs `.yml`, hence the rename hack I suggested.